pkgs/top-level/all-packages.nix

```
@@ -3687,6 +3687,8 @@ in
 
   gitfs = callPackage ../tools/filesystems/gitfs { };
 
+  gitin = callPackage gitAndTools.gitin;
```

```suggestion
  gitin = gitAndTools.gitin;
```

Otherwise, `nix-build -A gitin` fails with:

```
error: expression does not evaluate to a derivation (or a set or list of those)
```

---

I just added this change too.

I suppose adding to this line is deprecated to keep diffs easier : 
https://github.com/NixOS/nixpkgs/blob/b3b8f66ecc49e9592ccba041519ed9afac9aa355/pkgs/top-level/all-packages.nix#L19666

---

I agree. Go for it.

---

Besides that all looks good.

---

Actually, I was trying to say one should _not_ add it there.

It looks like this was an old convention and all the packages on that line were added in 2015 or before.

After some more research on my side, I am led to believe that recent packages aren't added to all-packages.nix at all:

- https://github.com/NixOS/nixpkgs-channels/commit/6d5fddac24b0ac9bf4ddd165883e7daea10bdeb0
- https://github.com/NixOS/nixpkgs-channels/commit/a05d11a95ae2e5e7fc6f92a7445259bc53d26a6a
- https://github.com/NixOS/nixpkgs-channels/commit/5de5d753bab7bac3e07003443e26766fde39b197
- https://github.com/NixOS/nixpkgs-channels/commit/183e706767f1983dcd16e9119740a6647e73a687

Should I just remove my changes in all-packages.nix and rename this merge request to:
`gitAndTools.gitin: init at 0.27.8` ?

---

@doronbehar should I go for

1. `gitin = gitAndTools.gitin;` (current version)
1. `inherit (gitAndTools) git ... gitin`
1. don't add it to `all-packages.nix` and have it only usable with the full name : `gitAndTools.gitin`

I think it should be option 1. or 3.

I was trying to say in this comment : https://github.com/NixOS/nixpkgs/pull/86070#discussion_r416857739, that 2. should be avoided since :
- it makes diffs harder
- recent gitAndTools packages have not been added here

---

I would say go for 3, since that seems to be current practice for new packages - we can always add it to `all-packages.nix` later if that feels necessary. It will still appear in searches, so we're not losing much discoverability either. Does that make sense?

---

@talyz, yes, completely. I just updated the PR with those changes.